wandb/sdk/launch/github_reference.py

```diff
@@ -231,3 +241,48 @@ def _update_path(self, dst_dir: str) -> None:
         elif path.is_dir():
             self.directory = self.path
             self.path = None
+
+    def _clone_repo(self, dst_dir: Optional[str] = None) -> None:
+        """Clone the repo to the destination directory, or to a temp directory if not given.
+
+        :returns: the local directory the repo was cloned to
+        """
+        if dst_dir is None and self.local_dir is not None:
+            # Repo already cloned, location is stored in self.local_dir
+            return
+        import git
+
+        if dst_dir is None:
+            dst_dir = tempfile.TemporaryDirectory().name
+        self.repo_object = git.Repo.clone_from(self.repo_ssh(), dst_dir)
+        self.local_dir = dst_dir
+
+    def get_commit(self) -> str:
+        """Get git hash associated with the reference."""
+        self._clone_repo()
+        if self.repo_object is None:
+            raise LaunchError(f"Error cloning git repo: {self.repo_ssh()}")
+        return self.repo_object.head.commit.hexsha  # type: ignore
+
+    def get_requirements(self) -> str:
```

Comment:
IMO, this should return `Optional[str]` and just return None if the file DNE. And otherwise return a path to thre file. `get_root_requirements_file`